lib/service/Service.ts

```diff
@@ -356,6 +356,12 @@ class Service extends EventEmitter {
    * Subscribe to orders being added to the order book.
    */
   public subscribeAddedOrders = (callback: (order: Order) => void) => {
+    this.orderBook.pairIds.forEach((pair) => {
+      const ownOrders = this.orderBook.getOwnOrders(pair);
+      const peerOrders = this.orderBook.getPeersOrders(pair);
+      const orders = [...ownOrders.buy, ...ownOrders.sell, ...peerOrders.buy, ...peerOrders.sell];
```

Comment:
I think this line is a little bit expensive, basically creating a new object with potentially thousands of keys that we will only use in this scope. It would be more performant to call `foreach` on all four of these arrays separately.